src/tools/tidy/src/ui_tests.rs

```diff
@@ -11,35 +13,43 @@ const ROOT_ENTRY_LIMIT: usize = 948;
 const ISSUES_ENTRY_LIMIT: usize = 2117;
 
 fn check_entries(path: &Path, bad: &mut bool) {
-    let dirs = walkdir::WalkDir::new(&path.join("test/ui"))
-        .into_iter()
-        .filter_entry(|e| e.file_type().is_dir());
-    for dir in dirs {
-        if let Ok(dir) = dir {
-            let dir_path = dir.path();
+    for dir in Walk::new(&path.join("test/ui")) {
+        match dir {
+            Ok(entry) => {
```

Comment:
nit: changing this from `if let` to match makes the diff harder to understand

Reply:
Add `if let`, but seems we can not use `let_chains`?

```console
   = note: see issue #53667 <https://github.com/rust-lang/rust/issues/53667> for more information
   = help: add `#![feature(let_chains)]` to the crate attributes to enable
```

And I and it to `lib.rs` also report:
```console
error[E0725]: the feature `let_chains` is not in the list of allowed features
 --> src/tools/tidy/src/lib.rs:1:12
  |
1 | #![feature(let_chains)]
  |            ^^^^^^^^^^
```

Reply:
ah, I see. this is fine then